Content.Shared/Burial/BurialSystem.cs

```diff
@@ -51,8 +51,15 @@ private void OnInteractUsing(EntityUid uid, GraveComponent component, InteractUs
                 BreakOnHandChange = true
             };
 
+            if (component.Stream == null)
+                component.Stream = _audioSystem.PlayPredicted(component.DigSound, uid, args.User)?.Entity;
+
             if (!_doAfterSystem.TryStartDoAfter(doAfterEventArgs))
+            {
+                _audioSystem.Stop(component.Stream);
                 return;
+            }
```

Comment:
Shouldn't you just start it if the doafter succeeds? Seems weird to just start it and immediately delete it within 2 lines.

Reply:
That's what happened before, it would start if the doafter started and play until it finished. The thing is, because aghosts just bypass the doafter time the doafter system immediately calls the doafter event. This leads to the code handling a doafter finishing being unable to stop the sound because it does not yet exist. As such the sound will start immediately after never be stopped.

Reply:
I got an idea.